pkg/vsphere/vm/vm.go

```diff
@@ -198,30 +199,30 @@ func (vm *VirtualMachine) WaitForKeyInExtraConfig(ctx context.Context, key strin
 					if key == value.GetOptionValue().Key {
 						detail = value.GetOptionValue().Value.(string)
 						if detail != "" && detail != "<nil>" {
-							return true
+							exitEarly = true
```

Comment:
How does it help to iterate over all values? It literally looks like you just need to exist in place where you assign true.